include/uacpi/helpers.h

```diff
@@ -1,6 +1,8 @@
 #pragma once
 
-#ifdef __cplusplus
+#if defined(__WATCOMC__)
```

Comment:
I guess it's time to move `UACPI_STATIC_ASSERT` to `compiler.h` and make it a no-op for Watcom, I remember some versions or flags of MSVC having issues with the `msg` part of static assert as well so it's been long time coming.

Really hope we can keep all compiler specific code in platform/